### PR TITLE
Dev/peefy/upgrade kcl code

### PR DESCRIPTION
**Pull request by Peefy**
Purpose:
1. upgrade kcl code
2. bump kclvm expected version to v0.4.2

### CODE REVIEW

File: appops/nginx-example/dev/ci-test/stdout.golden.yaml [hunk 1]
```diff
@@ -1,8 +1,3 @@
-id: apps/v1:Deployment:nginx-example:nginx-exampledev
```
Comment (howieyuen) [inline appops/nginx-example/dev/ci-test/stdout.golden.yaml:1]: changes should not appears here. the newest version of kusion is equal to `kcl compile`

**Review by howieyuen — CHANGES_REQUESTED** [review]

File: base/pkg/kusion_models/kube/backend/job_backend.k [hunk 2]
```diff
@@ -5,6 +5,15 @@ import base.pkg.kusion_models.kube.utils
 import base.pkg.kusion_models.kube.metadata
 import base.pkg.kusion_models.kube.resource
 
+# Frontend volume to kubernetes volume mapping.
```
[the remark this answers is not in the document]
Comment (Peefy) [reply base/pkg/kusion_models/kube/backend/job_backend.k:8]: The latest version of KCL may not support forward references to global variables like this:
````
schema A:
    b: int = a

a = 1
````
So change it to
````
a = 1
schema A:
    b: int = a
````

File: appops/nginx-example/dev/ci-test/stdout.golden.yaml (hunk 1 above)
Comment (Peefy) [reply appops/nginx-example/dev/ci-test/stdout.golden.yaml:1]: I do use the latest version of kusion compile, attributes will show up at the top, but it doesn't affect Config CI tests

**Comment by howieyuen** [comment]
please squash commits to be compatible with purpose of this PR.

**Pull request closed without merging**